lib/internal/url.js

```diff
@@ -1436,6 +1436,10 @@ function toPathIfFileURL(fileURLOrPath) {
   return fileURLToPath(fileURLOrPath);
 }
 
+function getURLOrigin(url) {
```

Comment:
```suggestion
/**
 * This util takes a string containing a URL and return the URL origin,
 * its meant to avoid calls to `new URL` constructor.
 * @param {string} url
 * @returns {URL['origin']}
 */
function getURLOrigin(url) {
```
I don't have a suggestion for the method name.